src/utils/calldata/requestParser.ts

```diff
@@ -0,0 +1,122 @@
+import { AbiEntry, ParsedStruct, Tupled, abiStructs } from '../../types';
+import { BigNumberish } from '../number';
+import { isText, splitLongString } from '../shortString';
+import { felt, isTypeArray, isTypeFeltArray, isTypeStruct, isTypeTuple } from './cairo';
+import extractTupleMemberTypes from './tuple';
+
+/**
+ * Parse tuple type string to array of known objects
+ * @param element request element
+ * @param typeStr tuple type string
+ * @returns Tupled[]
+ */
+function parseTuple(element: object, typeStr: string): Tupled[] {
+  const memberTypes = extractTupleMemberTypes(typeStr);
+  const elements = Object.values(element);
+
+  if (elements.length !== memberTypes.length) {
+    throw Error(
+      `ParseTuple: provided and expected abi tuple size do not match.
+      provided: ${elements} 
+      expected: ${memberTypes}`
+    );
+  }
+
+  return memberTypes.map((it: any, dx: number) => {
+    return {
+      element: elements[dx],
+      type: it.type ?? it,
+    };
+  });
+}
+
+/**
+ * Deep parse of the object that has been passed to the method
+ *
+ * @param element - element that needs to be parsed
+ * @param type  - name of the method
+ * @param structs - structs from abi
+ * @return {string | string[]} - parsed arguments in format that contract is expecting
+ */
+function parseCalldataValue(
+  element: ParsedStruct | BigNumberish | BigNumberish[],
+  type: string,
+  structs: abiStructs
+): string | string[] {
+  if (element === undefined) {
+    throw Error(`Missing parameter for type ${type}`);
+  }
+  if (Array.isArray(element)) {
+    throw Error(`Array inside array (nD) are not supported by cairo. Element: ${element} ${type}`);
+  }
+  // checking if the passed element is struct
```

Comment:
in general, like my two other comments, is just to think about if we can extract some of these portions into a separate function, remove these comments and name the function similar to the comment! 
Maybe it's overkill or unnecessary for most, but just to think about it

Reply:
Regarding all splitting  i think we can do additional module cleanup post cari 1 changes when we can do adaptations for cairo1